analyzer/src/main/kotlin/managers/utils/NuGetSupport.kt

```diff
@@ -43,296 +56,278 @@ import org.ossreviewtoolkit.model.Scope
-                    nodeAsPair.second
-                )
-
-                subPackageRef?.let { packageReference.dependencies += it }
-            }
         }
-
-        return packageReference
-    }
-
-    private fun preparePackageReference(packageID: String, version: String): JsonNode? {
-        val (nuspecUrl, catalogUrl) = getInformationURL(packageID, version) ?: return null
-        val packageJsonNode = jsonMapper.readTree(catalogUrl.requestFromNugetAPI()) ?: EMPTY_JSON_NODE
-
-        packageReferencesAlreadyFound[Pair(packageID, version)] = Pair(nuspecUrl, packageJsonNode)
-
-        return packageJsonNode
     }
 
-    private fun getInformationURL(packageID: String, version: String): Pair<String, String>? {
-        val registrationInfo = try {
-            "https://api.nuget.org/v3/registration3/$packageID/$version.json".requestFromNugetAPI()
-        } catch (e: IOException) {
+    fun buildDependencyTree(
+        references: Collection<Identifier>,
+        dependencies: MutableCollection<PackageReference>,
+        packages: MutableCollection<Package>,
+        issues: MutableCollection<OrtIssue>
+    ) {
+        references.forEach { id ->
             try {
-                getIdUrl(packageID, version).requestFromNugetAPI()
-            } catch (e: IOException) {
-                ""
-            }
-        }
-
-        val node = jsonMapper.readTree(registrationInfo)
-
-        return if (node?.isMissingNode == false) {
-            Pair(getNuspecURL(node), getCatalogURL(node))
-        } else {
-            null
-        }
-    }
-
-    private fun getIdUrl(packageID: String, version: String) =
-        getCreateSearchRestAPIURL(packageID).let { node ->
-            getRightVersionUrl(node["data"]?.elements(), packageID, version)
-                ?: getFirstMatchingIdUrl(node["data"]?.elements(), packageID).orEmpty()
-        }
+                val (all, pkg) = packageCache.getOrPut(id) {
+                    val all = getAllPackageData(id)
+                    all to getPackage(all)
+                }
 
-    private fun getRightVersionUrl(
-        dataIterator: Iterator<JsonNode>?,
-        packageID: String, version: String
-    ): String? {
-        if (dataIterator == null) return null
-
-        while (dataIterator.hasNext()) {
-            val packageNode = dataIterator.next()
-            if (packageNode["id"].textValueOrEmpty() == packageID) {
-                packageNode["versions"].elements().forEach {
-                    if (it["version"].textValueOrEmpty() == version) {
-                        return it["@id"].textValueOrEmpty()
-                    } else if (!dataIterator.hasNext() && version == "latest") {
-                        return it["@id"].textValueOrEmpty()
+                val pkgRef = pkg.toReference()
+                dependencies += pkgRef
+
+                // As NuGet dependencies are very repetitive, truncate the tree at already known branches to avoid it to
```

Comment:
Is this truncation actually new?

Reply:
No, it was just done way more "hidden" at https://github.com/oss-review-toolkit/ort/blob/20729b0f8896ef87a63532d044d5c49ba4246a99/analyzer/src/main/kotlin/managers/utils/NuGetSupport.kt#L216-L223